client/src/controllers/EnableDirtyFormCheckController.ts

```diff
@@ -0,0 +1,190 @@
+import { Controller } from '@hotwired/stimulus';
+
+export class EnableDirtyFormCheckController extends Controller<HTMLFormElement> {
+  static values = {
+    confirmationMessage: { default: ' ', type: String },
+    alwaysDirty: { default: false, type: Boolean },
+    isCommentApp: { default: true, type: Boolean },
+  };
+
+  declare confirmationMessageValue: string;
+  declare alwaysDirtyValue: boolean;
+  declare isCommentAppValue: boolean;
+
+  declare isDirtyTimeout: NodeJS.Timeout | number;
+  declare isCommentsDirtyTimeout: NodeJS.Timeout | number;
+  declare commentApp: any;
+  declare isDirty: boolean;
+  declare isCommentsDirty: boolean;
+  declare initialData: Map<string, (string | File)[]>;
+  declare formSubmitted: boolean;
+  declare observer: MutationObserver;
+
+  toggleFormSubmitted() {
+    this.formSubmitted = true;
+  }
+
+  isFormDirty() {
+    if (this.isDirty) {
+      return true;
+    }
+    if (!this.initialData) {
+      return false;
+    }
+
+    const formData = new FormData(this.element);
+    const keys = Array.from(formData.keys()).filter(
+      (key) => !key.startsWith('comments-'),
+    );
+    if (keys.length !== this.initialData.size) {
+      return true;
+    }
+
+    return keys.some((key) => {
+      const newValue = formData.getAll(key);
+      const oldValue = this.initialData.get(key);
+      if (newValue === oldValue) {
+        return false;
+      }
+      if (Array.isArray(newValue) && Array.isArray(oldValue)) {
+        return (
+          newValue.length !== oldValue.length ||
+          newValue.some((value, index) => value !== oldValue[index])
+        );
+      }
+      return false;
+    });
+  }
+
+  initialize(): void {
```

Comment:
```suggestion
  initialize() {
```
No need for the void declaration here. 

Also, it's a good idea to put this method above all other methods (after declarations) as it's easier to follow the code flow this way. 

Curious, any reason you went for initialise and not connect?

Reply:
hey there were some bugs while i was testing with `connect()` when i use `initialize()` it worked correctly